pyiceberg/io/fsspec.py

```diff
@@ -168,6 +169,9 @@ def _s3(properties: Properties) -> AbstractFileSystem:
     if request_timeout := properties.get(S3_REQUEST_TIMEOUT):
         config_kwargs["read_timeout"] = float(request_timeout)
 
+    if _force_virtual_addressing := properties.get(S3_FORCE_VIRTUAL_ADDRESSING):
+        config_kwargs["s3"] = {"addressing_style": "virtual"}
```

Comment:
👍 
`config_kwargs` is passed to `botocore.client.Config`
https://s3fs.readthedocs.io/en/latest/api.html#s3fs.core.S3FileSystem

`botocore.client.Config` expects `botocore.client.Config` in `s3` dict
https://botocore.amazonaws.com/v1/documentation/api/latest/reference/config.html